contracts/cosmwasm-vm/cw-xcall-ibc-connection/src/contract.rs

```diff
@@ -217,6 +219,30 @@ impl<'a> CwIbcConnection<'a> {
             QueryMsg::GetUnclaimedFee { nid, relayer } => {
                 to_binary(&self.get_unclaimed_fee(deps.storage, nid, relayer))
             }
+            QueryMsg::GetIBCConfig { nid } => {
+                let ibc_config = self.get_ibc_config(deps.storage, &nid).unwrap();
+                let channel_config = self
+                    .get_channel_config(deps.storage, &ibc_config.src_endpoint().channel_id)
+                    .unwrap();
+                let config = HashMap::from([
+                    (
+                        "channelId",
+                        ibc_config.src_endpoint().channel_id.to_string(),
+                    ),
+                    ("port", ibc_config.src_endpoint().port_id.to_string()),
+                    (
+                        "destinationChannelId",
+                        ibc_config.dst_endpoint().channel_id.to_string(),
+                    ),
+                    (
+                        "destinationPort",
+                        ibc_config.dst_endpoint().port_id.to_string(),
+                    ),
+                    ("lightClient", channel_config.client_id),
+                    ("timeoutHeight", channel_config.timeout_height.to_string()),
+                ]);
+                to_binary(&config)
+            }
```

Comment:
instead of using hashmap we can declare a new response type :
`#[cw_serde]
ConfigResponse {
ibc_config:IbcConfig,
channel_config:ChannelConfig
}`

Reply:
bot ibc config and channel config are already serializable.

Reply:
But we might want to keep the json same in java and cosmwasm?

Reply:
ya that would be nice practice.